### PR TITLE
fix: pass proper grading status during ags recalculate

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/INF-183

## What's Changed

- Fixed change of gradingProgress during AGS call

## Dependencies PRs

- https://github.com/oat-sa/extension-tao-outcome/pull/242
- https://github.com/oat-sa/extension-tao-ltideliveryprovider/pull/334

## How to test
-Create a test with an item with the SCORE variable manually graded.
-Take the test. The item should have SCORE = 0.
-Hit the API in TR-4956 with send_ags parameter set to true and no request body.
- Check the payload of the AGS notification.

### CODE REVIEW

File: models/QtiRunnerInitDataBuilder.php [hunk 1]
```diff
@@ -123,6 +123,34 @@ public function build($deliveryExecutionId, $withScores = false)
         return $init;
     }
 
+    public function getQtiTestItems(string $deliveryExecutionId): array
```
Comment (andreluizmachado) [inline models/QtiRunnerInitDataBuilder.php:126]: Can we add unit tests here?

File: models/QtiRunnerInitDataBuilder.php [hunk 2]
```diff
@@ -123,6 +123,34 @@ public function build($deliveryExecutionId, $withScores = false)
...
+    {
+        $context = $this->getServiceContext($deliveryExecutionId);
+        $testDefinition = taoQtiTest_helpers_Utils::getTestDefinition($context->getTestCompilationUri());
+        $testParts = [];
+        foreach ($testDefinition->getTestParts() as $testPart) {
+            $sections = [];
+            foreach ($testPart->getAssessmentSections() as $section) {
+                $items = [];
+                foreach ($section->getSectionParts() as $item) {
+                    $itemData = $this->qtiRunnerService->getItemData($context, $item->getHref());
```
Comment (pnal) [inline models/QtiRunnerInitDataBuilder.php:136]: I'm afraid we can not use this extension code for this task, we have to somehow use qtiRunnerService directly instead, cause adding new dependency on extension-lti-test-review will be totally breaking change.

**Review by pnal — CHANGES_REQUESTED** [review]

File: models/QtiRunnerInitDataBuilder.php (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply models/QtiRunnerInitDataBuilder.php:136]: I've moved this logic here to tao-outcome repo here https://github.com/oat-sa/extension-tao-outcome/pull/242/files#diff-21357e0b8799d4075614d38401cd14f2e765bbab11421ceac475cedb3b43e7d7 so I will close this PR

File: models/QtiRunnerInitDataBuilder.php (hunk 1 above)
Comment (Karol-Stelmaczonek) [reply models/QtiRunnerInitDataBuilder.php:126]: I've moved this logic here https://github.com/oat-sa/extension-tao-outcome/pull/242/files#diff-21357e0b8799d4075614d38401cd14f2e765bbab11421ceac475cedb3b43e7d7 and created test here https://github.com/oat-sa/extension-tao-outcome/pull/242/files#diff-6ffe142eaddf3b73c81cffe82809d046f3a37228737e99f6a60a492c7ca97fbc

**Pull request closed without merging**